### PR TITLE
Add tasks to goal

### PR DESCRIPTION
**Pull request by nthngoc277**
Allow adding multiple tasks to goal when a goal is created

### CODE REVIEW

**Comment by nthngoc277** [comment]
@wilgoszpl I am adding a new feature which allows the user to add tasks for a goal. The server side works well right now, you can take a look on it, about the front end, I am trying to figure out why creating a goal with tasks still not work, although it works when there are no tasks.

**Comment by nthngoc277** [comment]
@wilgoszpl I have fixed the issue from front-end, now the feature works well ^_^
The thing I learned is that custom type from Backend can call directly from `gql` of front-end.

File: server/db/schema.rb [hunk 1]
```diff
@@ -26,6 +26,15 @@
     t.index ["user_id"], name: "index_goals_on_user_id"
   end
 
+  create_table "tasks", force: :cascade do |t|
+    t.string "description"
+    t.bigint "goal_id"
```
Comment (wilgoszpl) [inline server/db/schema.rb:31]: I'm a fan of using `uuid` instead bigint. Uuid is also unique, but it prevents from guesing how many records you have in your system and scrap through them.

Comment (nthngoc277) [reply server/db/schema.rb:31]: @wilgoszpl I used the generated code from `rails g` with `goal:references` and did not notice at this one until I see your comment. 

After searching about uuid vs bigint. I found this interesting post https://tomharrisonjr.com/uuid-or-guid-as-primary-keys-be-careful-7b2aa3dcb439 which is showing the pros and cons of both ways, and how he implemented both in his project.

How do you think about it?

**Pull request merged**